### ISSUE TITLE
Archive Title & Term Description – Display data when possible

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#44158]
With https://github.com/WordPress/gutenberg/pull/43699 merged, it would be nice to apply the same treatment to the Archive Title and Term Description blocks.

At the moment, if you create a template for a specific category, the Query Loop within that template will display posts from that category.

However, Archive Title and Term Description continue to display their generic placeholders:

<img width="1147" alt="Screenshot 2022-09-14 at 14 59 56" src="https://user-images.githubusercontent.com/846565/190174923-b54f15a9-3c31-4cd3-aafa-34712d2f5d9d.png">

Since the template infers context, ideally these blocks should behave like Query Loop and display the actual title / description.

cc @oandregal since you worked on this before :)

**Comment by oandregal** [comment]
For the template of a `base` category I understand we want to show `Category: base` in the archive title. What should we do for the fallback category template (any category)?

**Comment by jameskoster** [comment]
Something like this makes sense to me:

| Template | Archive Title renders |
| --- | --- |
| category.html | Category: Name |
| category-recipes.html | Category: Recipes | 

This scheme can be used with Tags and other archives too.

What do you think?

**Comment by annezazu** [comment]
Can we close this with https://github.com/WordPress/gutenberg/pull/52521 merged?

**Comment by jameskoster** [comment]
Not quite. We still need to handle date archives which are a bit tricky as the placeholder needs to account for all the different formats, e.g.:

> Year: 2023
> Month: May 2023
> Day: May 10, 2023

Perhaps something like `Year/Month/Day: Value`?
